SwiftGit2/Repository.swift

```diff
@@ -537,4 +537,137 @@ final public class Repository {
 		let iterator = CommitIterator(repo: self, root: branch.oid.oid)
 		return iterator
 	}
+
+	// MARK: - Diffs
+
+	public func diff(for commit: Commit) -> Result<[Diff.Delta], NSError> {
+		return self
+			.commit(with: commit.oid)
+			.flatMap { baseCommit in
+				return self.tree(from: baseCommit)
+			}
+			.flatMap { baseTree in
+				guard !commit.parents.isEmpty else {
+					// Initial commit in a repository
+					return self.diff(fromTree: nil, toTree: baseTree)
+				}
+
+				let mergeDiff: Result<OpaquePointer?, NSError> = .success(nil)
+				for parent in commit.parents {
+					let diff = self
+						.commit(with: parent.oid)
+						.flatMap { commit in
+							return self.tree(from: commit)
+						}
+						.flatMap { tree in
+							return self.diff(fromTree: tree, toTree: baseTree)
+						}
+					return mergeDiff
```

Comment:
This doesn't look correct. If it's a hard `return` here, then it doesn't look like it will work for merge commits (commits with >1 parent).

That means we should add a test for this case.